Source/VungleAdapterBannerAd.swift

```diff
@@ -5,268 +5,150 @@
-        adWasDisplayed = true
     }
-    
-    private func finishDisplayingVungleAd() {
-        // Programmatic
-        if let adm = request.adm {
-            VungleSDK.shared().finishDisplayingAd(request.partnerPlacement, adMarkup: adm)
-        } else {
-        // Non-programmatic
-            VungleSDK.shared().finishDisplayingAd(request.partnerPlacement)
-        }
-    }
-    
+
     /// Mapping of the request ad size to Vungle's ad size type. Nil means a MREC banner.
-    private var vungleAdSize: VungleAdSize? {
-        let size = request.size ?? IABStandardAdSize
-        switch size {
-        case IABMediumAdSize:
-            // Note that Vungle MREC ads are not considered banners by Vungle and require using a different set of methods.
-            // We return nil here to signify that this is not a Vungle banner, but an MREC ad.
-            return nil
+    private var vungleAdSize: BannerSize {
+        switch request.size {
+        case IABStandardAdSize:
+            return BannerSize.regular
+        case CGSize(width: 300, height: 50):
+            return BannerSize.short
         case IABLeaderboardAdSize:
-            return .bannerLeaderboard
+            return BannerSize.leaderboard
+        case IABMediumAdSize:
+            return BannerSize.mrec
         default:
-            return .banner
-        }
-    }
-    
-    /// A size for the Vungle banner/MREC ad container compatible with that specific Vungle ad type.
-    /// Note that this size must match Vungle's expectations or the ad rendering will fail.
-    /// See Vungle's documentation on banner ads: https://support.vungle.com/hc/en-us/articles/360048572771
-    /// See Vungle's documentation on MREC ads: https://support.vungle.com/hc/en-us/articles/360048079292
-    private var adContainerSize: CGSize {
-        switch vungleAdSize {
-        case nil:
-            // A nil vungleAdSize means the ad is not technically a Vungle banner, but a Vungle MREC ad.
-            // MREC ads must always have a size 300x250 per Vungle's documentation.
-            return IABMediumAdSize
-        case .banner:
-            return IABStandardAdSize
-        case .bannerShort:
-            return CGSize(width: 300, height: 50)
-        case .bannerLeaderboard:
-            return IABLeaderboardAdSize
-        case .unknown:
-            return IABStandardAdSize
-        @unknown default:
-            return IABStandardAdSize
+            log("Unrecognized banner size.")
+            // Ad is unlikely to load if the size doesn't match, but we need to return something
+            return BannerSize.regular
         }
     }
 }
 
-// MARK: - VungleSDKHBDelegate
+// MARK: - VungleBannerDelegate
+extension VungleAdapterBannerAd: VungleBannerDelegate {
+    func bannerAdDidLoad(_ banner: VungleBanner) {
+
+        // Check that the ad is ready
+        guard banner.canPlayAd() == true else {
+            let error = error(.showFailureAdNotReady)
```

Comment:
nit: seems a bit off that other errors are loadFailure's, and we are reporting back a load result, but here we use a showFailure. Maybe there's some loadFailure code that could suit us? If not it's though, maybe loadFailureUnknown with a custom description could help